server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+                logger.debug("Latest image received: " + latest.toString());
+                imageId = latest.getImageId();
+            } else {
+                logger.warn("No images received defaulting to old ubuntu 16.04 image");
+                imageId = "ami-0def3275";
+                //TODO maybe not defaulting but throwing a transformation failed exception?
+            }
+        } catch (SdkClientException se) {
+            logger.error("Cannot connect to AWS to request image Ids, defaulting to old ubuntu 16.04 image");
+            imageId = "ami-0def3275";
+            //TODO maybe not defaulting but throwing a transformation failed exception?
+        }
+        logger.debug("ImageId is: " + imageId);
+        return imageId;
+    }
+
+    /**
+     Finds the best InstanceType based on the values contained in the ComputeCapability.
+     If necessary the values are scaled upwards till they meet the requirement.
+
+     @param computeCapability The ComputeCapability to map.
+     @param distinction       A distinction string. Can either be "EC2" or "RDS".
+     @return A valid InstanceType / InstanceClass string.
+     @throws TransformationFailureException Gets thrown if the values numCpus and memSize are too big and there is no
+     valid InstanceType.
+     */
+    public String mapComputeCapabilityToInstanceType(ComputeCapability computeCapability, String distinction) throws
+        TransformationFailureException {
+        //TODO what to do with disk size?
+        Integer numCpus = computeCapability.getNumCpus().orElse(0);
+        Integer memSize = computeCapability.getMemSizeInMB().orElse(0);
+        //default type the smallest
+        final ImmutableList<InstanceType> instanceTypes;
+        if ("EC2".equals(distinction)) {
+            instanceTypes = EC2_INSTANCE_TYPES;
+        } else if ("RDS".equals(distinction)) {
+            instanceTypes = RDS_INSTANCE_CLASSES;
+        } else {
+            throw new IllegalArgumentException("Distinguisher not supported");
+        }
+        List<Integer> allNumCpus = instanceTypes.stream()
+            .map(InstanceType::getNumCpus)
+            .sorted()
+            .collect(Collectors.toList());
+        List<Integer> allMemSizes = instanceTypes.stream()
+            .map(InstanceType::getMemSize)
+            .sorted()
+            .collect(Collectors.toList());
+        // scale numCpus and memSize upwards if they are not represented in the lists
+        try {
+            logger.debug("Check numCpus: " + numCpus);
+            numCpus = checkValue(numCpus, allNumCpus);
+            logger.debug("Check memSize: " + memSize);
+            memSize = checkValue(memSize, allMemSizes);
+        } catch (IllegalArgumentException ie) {
+            String errorMessage = "Values numCpus: " + numCpus + " and/or memSize: " + memSize + " are to big. No " +
+                "InstanceType found";
+            logger.error(errorMessage);
```

Comment:
I think it is sufficient to add the errorMessage to the `TransformationFailureException`